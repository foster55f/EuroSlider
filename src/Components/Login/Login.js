import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';


const Login = (props) => {

const goToFavorites = () => {
    props.history.push(`/yourfavorites`)
    }  

        return (
            <>
            <div className='header-background'>
                <nav>
                <h2 className='header-title'>Euro Slider</h2>
                        <>
                            <button className='favorite-button' onClick={goToFavorites}> Click to View Your Favorite Highlights!</button>
                        </>
                </nav>
            </div>
            </>
        )
    }

export default (withRouter(Login))

Login.propTypes = {
    user: PropTypes.object,
    userLog: PropTypes.bool,
    addUser: PropTypes.func,
    userLogIn: PropTypes.func,
}