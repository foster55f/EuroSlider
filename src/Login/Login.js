import React from 'react';
import { withRouter } from 'react-router-dom';


class Login extends React.Component {
    constructor() {
        super();
        this.state = {
            
        }
    }

    login = () => {
    this.props.history.push(`/yourFavorites`)
        // this.setState({ loggedIn: true })
        // if (this.state.username === '' || this.state.passwordLength > 0) {
        //     this.setState({ error: 'THE USERNAME OR PASSWORD IS INCORECT' })
        //   }
        // localStorage.setItem("test1", this.state.username[0])
        // localStorage.getItem("test1", this.state.username)
    }

    render() {
        return (
            <>
            <div className='header-background'>
                <nav>
                <h2 className='header-title'>Euro Slider</h2>
                            <>
                            <button onClick={this.login}> Euro Vids To Share With Friends!</button> 
                            </>
                </nav>
            </div>
            </>
        )
    }
}

export default withRouter(Login)