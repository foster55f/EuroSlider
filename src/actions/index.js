export const addGames = games => ({
    type: 'ADD_GAMES',
    games
})

export const filterGames = displayGames => ({
    type: 'FILTER_GAMES',
    displayGames
})

