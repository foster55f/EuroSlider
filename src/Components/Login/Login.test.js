import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { addUser, userLogIn } from '../../actions';


describe('Login', () => {
  let wrapper;

  it('should match the App snapshot', () => {
    wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  })


  it('should update state on change', () => {
    let mockEvent = { target: { value: 'foster' } }
    let initialState = { username: '' }
    let expectedState = 'foster' 
    
    wrapper.setState(initialState)
    wrapper.find('.userName').simulate('change', mockEvent)
    expect(wrapper.state('username')).toEqual(expectedState)
  });

  it('should run goToUserFavorites when the button is clicked', () => {
    wrapper.instance().goToUserFavorites = jest.fn();
    wrapper.instance().goToUserFavorites();
    const mockEvent = { preventDefault: jest.fn() };
  
    wrapper.find('.favorite-button').simulate('click', mockEvent);
  
    expect(wrapper.instance().goToUserFavorites).toHaveBeenCalled();
  });

  it('should run goToFavorites when the button is clicked', () => {
    wrapper.instance().goToFavorites = jest.fn();
    wrapper.instance().goToFavorites();
    const mockEvent = { preventDefault: jest.fn() };
  
    wrapper.find('.favorite-button').simulate('click', mockEvent);
  
    expect(wrapper.instance().goToFavorites).toHaveBeenCalled();
  });


  describe('mapsStateToProps', () => {
    it('should return only userinfo and user properties from the store', () => {
        const mockState = {
            user: {},
            userLog: false,
        };
        const expected = {
            user: {},
            userLog: false
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
    });

});

describe('mapDispatchToProps', () => {
    it('calls dispatch with an addUser action when addUser is called',
        () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = addUser({ sample: 'user' });
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.addUser({ sample: 'user' });

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });
  
      it('calls dispatch with an userLogin action when userLogin is called',
      () => {
          const mockDispatch = jest.fn();
          const actionToDispatch = userLogIn(true);
          const mappedProps = mapDispatchToProps(mockDispatch);

          mappedProps.userLogIn(true);

          expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });  
})