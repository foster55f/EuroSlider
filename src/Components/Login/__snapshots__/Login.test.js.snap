// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the App snapshot 1`] = `
<Fragment>
  <div
    className="header-background"
  >
    <nav>
      <h2
        className="header-title"
      >
        Euro Slider
      </h2>
      <div
        className="user-entry"
      >
        <input
          className="userName"
          onChange={[Function]}
          placeholder="ENTER NAME"
          type="text"
        />
        <button
          className="create-user-button"
          onClick={[Function]}
        >
           Click to Create Your Profile
        </button>
      </div>
      <button
        className="favorite-button"
        onClick={[Function]}
      >
         Click to View Your Favorite Highlights!
      </button>
    </nav>
  </div>
</Fragment>
`;
